python/ray/autoscaler/command_runner.py

```diff
@@ -440,14 +446,14 @@ def _shutdown_after_next_cmd(self):
         self.shutdown = True
 
     def _check_container_status(self):
-        no_exist = "not_present"
-        cmd = check_docker_running_cmd(self.docker_name) + " ".join(
-            ["||", "echo", quote(no_exist)])
+        if self.initialized:
+            return True
         output = self.ssh_command_runner.run(
-            cmd, with_output=True).decode("utf-8").strip()
-        if no_exist in output:
-            return False
-        return "true" in output.lower()
+            check_docker_running_cmd(self.docker_name),
+            with_output=True).decode("utf-8").strip()
+        # Checks for the false positive where "true" is in the container name
+        return "true" in output.lower(
+        ) and "No such object" not in output.lower()
```

Comment:
```suggestion
        return ("true" in output.lower() and "No such object" not in output.lower())
```
wrapping the whole thing in parens should give better formatting